src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { ChangeEvent, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterType>(FilterType.all);
+  const [query, setQuery] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const handleFilterChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setFilter(event.target.value as FilterType);
+  };
+
+  const handleQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const queryCleaner = () => {
+    setQuery('');
+  };
```

Comment:
The `queryCleaner` function name is not very descriptive. It would be better to rename it to something like `handleQueryClear` to make it more consistent with other handlers and more understandable.